src/board/devices/input/rotarySwitch.ts

```diff
@@ -55,7 +55,20 @@ export class RotarySwitch extends Device {
    *
    * @returns: the current value of the rotary switch
    */
-  public getRotaryValue(): Byte {
-    return this.memory.readByte(RotarySwitch.ROTARYSWITCH_ADDRESS)
+  public getRotaryValue(): number {
+    if (
+      this.memory.readByte(RotarySwitch.ROTARYSWITCH_ADDRESS).value - 240 <
+      0
+    ) {
+      this.initializeByte()
+    }
+    return this.memory.readByte(RotarySwitch.ROTARYSWITCH_ADDRESS).value - 240
+  }
+
+  private initializeByte(): void {
```

Comment:
`InitializeByte` should rather be called from the constructor.

Reply:
Changed this